tsdb/engine/tsm1/reader.go

```diff
@@ -607,7 +607,7 @@ func (d *indirectIndex) UnmarshalBinary(b []byte) error {
 		i++
 
 		// 2 byte count of index entries
-		count := btou16(b[i : i+indexCountSize])
+		count := int32(btou16(b[i : i+indexCountSize]))
```

Comment:
OK, I see it now. `28` is shoved into 16 bits too.

The change is a little obscure. Would it be better to move the casting to nearer where it matters? That would probably make it clearer to the next guy.

I.e.

`i += int32(count) - 1 * int32(indexEntrySize))`

You get the idea -- deal with the overflow at the point where it matters.